paddle/fluid/framework/operator.cc

```diff
@@ -97,6 +97,15 @@ static LoD GetLoD(const Scope& scope, const std::string& name) {
   }
 }
 
+static bool InplaceConvWeights(const std::string& name,
+                               const OperatorWithKernel& op) {
+#ifdef PADDLE_WITH_MKLDNN
+  return name.find("conv2d") == 0 && name.find("GRAD") == std::string::npos &&
+         op.Attr<bool>("is_test");
+#endif
+  return false;
+}
+
```

Comment:
It's not appropriate adding conv function at operator.cc.

Reply:
The problem is that we need to reorder conv weights every time we use it and it's costly. That's because we cannot reorder conv weights in persistent way. Before entering convolution, weights are transformed to mkldnn format, but this tensor is only temporary. To change that, I've added exception in operator.cc and transform weights in place. I know it's not clean, but the only alternative I see in current architecture is to duplicate weights in device context. But that way we would unnecessarily consume more resources - possibly a lot of memory.

Reply:
>  It's not appropriate adding conv function at operator.cc

I agree with it. @Superjomn Do you have some suggestions about where to add conv function?